subcommand/server-acl-init/command_test.go

```diff
@@ -1505,43 +1064,31 @@ func TestRun_HTTPS(t *testing.T) {
 	caFile, certFile, keyFile, cleanup := generateServerCerts(t)
 	defer cleanup()
 
-	agentConfig := fmt.Sprintf(`
-		primary_datacenter = "dc1"
-		acl {
-			enabled = true
-		}
-		ca_file = "%s"
-		cert_file = "%s"
-		key_file = "%s"`, caFile, certFile, keyFile)
-
-	// NOTE: We can't use testutil.TestServer for this test because the HTTP
-	// port can't be disabled (causes a seg fault).
-	a := &agent.TestAgent{
-		Name:   t.Name(),
-		HCL:    agentConfig,
-		UseTLS: true, // this also disables HTTP port
-	}
-
-	a.Start()
-	defer a.Shutdown()
+	srv, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
 
-	createTestK8SResources(t, k8s, a.HTTPAddr(), resourcePrefix, "https", ns)
+		c.CAFile = caFile
+		c.CertFile = certFile
+		c.KeyFile = keyFile
+	})
+	require.NoError(err)
+	defer srv.Stop()
 
 	// Run the command.
 	ui := cli.NewMockUi()
 	cmd := Command{
 		UI:        ui,
 		clientset: k8s,
 	}
-	cmd.init()
+
 	responseCode := cmd.Run([]string{
-		"-server-label-selector=component=server,app=consul,release=" + releaseName,
 		"-resource-prefix=" + resourcePrefix,
 		"-k8s-namespace=" + ns,
 		"-use-https",
 		"-consul-tls-server-name", "server.dc1.consul",
 		"-consul-ca-cert", caFile,
-		"-expected-replicas=1",
+		"-server-address=127.0.0.1",
```

Comment:
why aren't you using `strings.Split(srv.HTTPSAddr, ":")[0]` here?

Reply:
🤣 I don't know! Probably because ☕ hasn't kicked in ⛔ 🧠